app/models/user.rb

```diff
@@ -45,12 +46,27 @@ def send_password_reset
     UserMailer.forgot_password(id).deliver
   end
 
+  def recompute_credits
+    self.credits_count = self.credits.inject(0) { |sum, credit| sum += credit.amount }
+    save!
```

Comment:
1. if you're using bang methods, recompute_credits should also be named as recompute_credits!
2. Why self.credits?
3. Why not directly calling sum using query?